oeps/processes/oep-0055-project-maintainers.rst

```diff
@@ -0,0 +1,309 @@
+of each component and should be welcoming and available.
+
+Non-technical responsibilities of maintainers include:
+
+* Effective communication with community members, other maintainers,
+  and project leadership
+* Disciplined management to ensure that maintenance work is done
+* Enforcement of the project's standards
+* The ability to solve or escalate issues facing users of the
+  component
+
+Technical responsibilities of maintainers include:
+
+* Understanding the current component architecture and implementation
+* Having a technical vision for the evolution of the component
+* The ability to assess the appropriateness of proposed changes to or
+  extensions of the component
+* The role of the component in the overall platform and how it
+  interacts with other components
+
+It is acceptable for a component to be maintained by a team or group
+as long as that arrangement does not reduce the reachablility of the
+maintainer or reduce accountability.
+
+Prerequisites
+-------------
+In order to be a repository maintainer, you must
+
+* Be an active member of the Open edX community who has contributed to
+  the project in the last 6 months.  The types of contributions that
+  are relevant will match those defined in the `core contributor
+  program`_
+* Abide by the `Open edX Code of Conduct
+  <https://openedx.org/code-of-conduct/>`_
+* Be a core contributor for the repository in question who has
+  demonstrated excellence in the role of core contributor.  From
+  time-to-time, in the case of a strong candidate and project need,
+  core contributor status and maintainer status may be conferred at
+  the same time.
+
+Bootstrapping
+-------------
+In order to ensure an orderly roll-out of the formal maintainer role,
+existing 2U staff in the owner role will transition into the role of
+maintainer for the components that they own.
+
+Community members who are acting in informal maintainer roles, will
+also be asked to become named maintainers.
+
+A list of all components without maintainers will be created and
+published.
+
+An asynchronous maintainer scrum-of-scrums will be created as a Slack
+channel.
+
+Standards for measuring the quality of repositories will be rolled out
+across all repositories.  These standards can benefit from work done
+at edX to measure `repository health <https://github.com/openedx/edx-repo-health>`_.
+
+Becoming a Maintainer
+---------------------
+After the bootstrapping phase, the Open edX project will need
+processes for adding additional maintainers, replacing maintainers,
+and ensuring that new components are maintained.
+
+**This section should be congruent with the Core Contributor process**
+
+A new maintainer can be added in any of the following ways:
+
+* An existing maintainer can request that a new person become a
+  component maintainer either in conjunction with the existing
+  maintainer or replacing them.  If there is no objection from
+  existing maintainers, the role will be conferred.
```

Comment:
I should clarify that.  I think we should allow multiple maintainers, especially for big or complex components.  Thoughts?

Reply:
If the component is sufficiently complex to justify multiple maintainers, those maintainers form a group and should organize internally and present a single point of contact (like teams-as-maintainers above).

... However, subsequent lines here really sound like process descriptions for removing individuals (e.g. changing employment) and that does not translate well to the team-maintainer idea. I think either you revise this all to be specific to individuals (not my preference) or you keep this generic and refer to CC processes for handling individual access (this seems cleaner to me).

Reply:
I'm updating this to address comments, as I would really like this OEP to move along. I can't quite parse your comment, @hurtstotouchfire , and would love some more detail. Why is it wrong to refer both to individuals and groups, in your mind? Are you suggesting most of the document be rewritten to refer to groups? (I'm not sure what you mean by "generic"). By "CC processes" do you mean Core Contributor processes outlined in OEP-54? So a new maintainer would be nominated by an existing one, and other maintainers would weighin?

Reply:
In https://producingoss.com/en/producingoss-letter.pdf p 78 "Adding new maintainers", the author makes a case for adding new maintainers to *not* be a public process. However I wonder if his definition of maintainer is really congruent with ours. I am ambivalent, I could see a public or a private process working. I think it is important, regardless, that there is a backstop in case self-organization isn't working.